src/OpenRiaServices.Server/Framework/Data/DomainOperationEntry.cs

```diff
@@ -191,7 +191,7 @@ private bool CalculateRequiresValidation()
             return this._attributes[typeof(ValidationAttribute)] != null
                 || this.Parameters.Any(p =>
                     {
-                        if (p.Attributes[typeof(ValidationAttribute)] != null)
+                        if (p.HasValidationAttributeOnProperties || p.Attributes[typeof(ValidationAttribute)] != null)
```

Comment:
1. I think it might make sense to move the logic to a separate Property on the Parameter entry as you have done here, it will allow us to easily only call validation for parameters which requires attributes.
but I think all code (including the code below) which handles complex types should be moved as well.

Reply:
It seems the validation logic has special logic for "*complex types*" and "*entity types*" and the root cause may be that only *complex types* are currently checked here (a few rows down).

Ex. Here is some code which checks if a type is an entity type
https://github.com/OpenRIAServices/OpenRiaServices/blob/b281a36deb236a5a86afb42d3630aaa33495b3b7/src/OpenRiaServices.Server/Framework/Data/DomainService.cs#L1217

Maybe the existing code can be slightly modified so that it enters the if statement and check the metatype for entities as well? The code for *MetaType.RequiresValidation* should work at least as well for us, it should be a bit more complete (and complex) 


```csharp
                       Type elementType = TypeUtility.GetElementType(p.ParameterType);
                       if (not TypeUtility.IsPredefinedSimpleType?
//        or        if (complex or entity type (DomainServiceDescription.IsKnownEntityType(...)) )
                        {
                            MetaType metaType = MetaType.GetMetaType(elementType);
                            return metaType.RequiresValidation;
                        }

```

Reply:
After some further investigation I think the easiest solution would be to remove the problematic if statement from which causes validation to not be performed.

https://github.com/OpenRIAServices/OpenRiaServices/blob/b281a36deb236a5a86afb42d3630aaa33495b3b7/src/OpenRiaServices.Server/Framework/Data/DomainService.cs#L1202

We could remove the if statement but I think it should be safe to move it to https://github.com/OpenRIAServices/OpenRiaServices/blob/b281a36deb236a5a86afb42d3630aaa33495b3b7/src/OpenRiaServices.Client/Framework/ValidationUtilities.cs#L374
and return early from that method instead to keep the optimization.
 Since the method only checks for validationattributes, and complex objects which seems to match what CalculateRequiresValidation does today. 

Trying to "fix" *RequiresValidation* might be more complex, since we would have to take inheritance into account and it seems like Submit also checks the property and assumes that it will return false if it is only the entity (first parameter) which needs to be validated